client/src/components/app.jsx

```diff
@@ -1,15 +1,50 @@
 import React from 'react';
+import axios from 'axios';
 
 class App extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-
+      requestedProductId: '4',
+      productId: '',
+      productDetails: {
+        designer: '',
+        materials: [],
+        care: [],
+        Packaging: {
+          packagingName: '',
+          packagingDescription: '',
+          width: '',
+          height: '',
+          length: '',
+          weight: '',
+          packages: ''
+        }
+      },
+      images: []
     }
   }
+
+  componentDidMount() {
+    axios.get(`/api/products/${this.state.requestedProductId}`)
+      .then((response) => {
+        this.setState({productId: response.data.productId})
+        this.setState({productDetails: response.data.productDetails})
```

Comment:
Great job using a promise here! 

If there isn't a particular reason that you are using three setStates here then It might be more efficient and look cleaner to write just one setState

somthing like: 

`setState({
   productId: response.data.productId,
   productDetails: response.data.productDetails,
   images: response.data.images
})`